x-pack/plugins/security_solution/common/utils/time_type_value.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash/fp';
+import type { Unit } from '@kbn/datemath';
+
+export const getTimeTypeValue = (time: string): { unit: Unit; value: number } => {
+  const timeObj: { unit: Unit; value: number } = {
+    unit: 'ms',
+    value: 0,
+  };
+  const filterTimeVal = time.match(/\d+/g);
+  const filterTimeType = time.match(/[a-zA-Z]+/g);
```

Comment:
Is there some reason to match all letters here and check separately for `s`, `m`, `h` and `d` below? What if there is `time` string like `1abch` or `20mms`?

It could one regex like

```
/^(\d+)?(ms|s|m|h|d)?$/
```

to match strictly the expected format.

Reply:
Agree with Maxim here, with the additional doubt of why defaulting to `0ms` if the only accepted `Unit`s are actually `['s', 'm', 'h', 'd']`

Reply:
Same here.

I just moved this method from another file x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_creation/helpers.ts :-)

If I will have time before the release to address the concerns that you put, I will look into this as well.